snapcraft/internal/pluginhandler.py

```diff
@@ -650,15 +650,34 @@ def _migratable_filesets(fileset, srcdir):
     return snap_files, snap_dirs
 
 
+def _migrate_files_directories(snap_dirs, srcdir, dstdir,
+                               follow_symlinks=False):
+    for directory in snap_dirs:
+        src = os.path.join(srcdir, directory)
+        dst = os.path.join(dstdir, directory)
+        os.makedirs(dst, exist_ok=True)
+        try:
+            shutil.copystat(src, dst, follow_symlinks=follow_symlinks)
+        except FileNotFoundError:
+            logger.info("src not found: {!r}".format(src))
+            pass
+
+
 def _migrate_files(snap_files, snap_dirs, srcdir, dstdir, missing_ok=False,
                    follow_symlinks=False, fixup_func=lambda *args: None):
-    for directory in snap_dirs:
-        os.makedirs(os.path.join(dstdir, directory), exist_ok=True)
 
+    _migrate_files_directories(snap_dirs, srcdir, dstdir,
+                               follow_symlinks=follow_symlinks)
     for snap_file in snap_files:
         src = os.path.join(srcdir, snap_file)
         dst = os.path.join(dstdir, snap_file)
         os.makedirs(os.path.dirname(dst), exist_ok=True)
+        try:
+            shutil.copystat(src, dst, follow_symlinks=follow_symlinks)
+        except FileNotFoundError:
+            logger.info("src not found: {!r}".format(src))
+            pass
```

Comment:
There are 30+ tests that fail for files that aren't in src.  install/foo, install/bar, etc.  Do you think the tests are broken?  If so, I can chase that down and not catch that exception.  Does the same apply to the directory copystat above as well?